File: example/src/main/java/io/castle/android/sample/SampleApplication.java
/*
 * Copyright (c) 2017 Castle
 */

package io.castle.android.sample;

import android.app.Application;

import java.util.Arrays;
import java.util.List;

import io.castle.android.Castle;
import io.castle.android.CastleConfiguration;

public class SampleApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();

        List<String> baseURLAllowlist = Arrays.asList(new String[] { "https://api.castle.io/" });

        // Create configuration object
        CastleConfiguration configuration = new CastleConfiguration.Builder()
                .publishableKey("pk_btApAXqt1jpJtEARf1stsnvyov6czPmn")
                .screenTrackingEnabled(true)
                .debugLoggingEnabled(true)
                .baseURLAllowlist(baseURLAllowlist)
                .build();

        // Setup Castle SDK with provided configuration
        Castle.configure(this, configuration);
    }
}

File: example/src/main/java/io/castle/android/sample/SampleApplication.kt
/*
 * Copyright (c) 2020 Castle
 */
package io.castle.android.sample

import android.app.Application
import io.castle.android.Castle
import io.castle.android.CastleConfiguration
import java.util.*

class SampleApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        val baseURLAllowList = listOf("https://m.castle.io/")

        // Create configuration object
        val configuration = CastleConfiguration.Builder()
                .publishableKey("pk_CTsfAeRTqxGgA7HHxqpEESvjfPp4QAKA")
                .screenTrackingEnabled(true)
                .debugLoggingEnabled(true)
                .baseURLAllowList(baseURLAllowList)
                .build()

        // Setup Castle SDK with provided configuration
        Castle.configure(this, configuration)
    }
}